test/e2e/framework/job/wait.go

```diff
@@ -230,29 +252,29 @@ func isConditionTrue(j *batchv1.Job, condition batchv1.JobConditionType) bool {
 
 // WaitForJobGone uses c to wait for up to timeout for the Job named jobName in namespace ns to be removed.
 func WaitForJobGone(ctx context.Context, c clientset.Interface, ns, jobName string, timeout time.Duration) error {
-	return wait.PollUntilContextTimeout(ctx, framework.Poll, timeout, false, func(ctx context.Context) (bool, error) {
-		_, err := c.BatchV1().Jobs(ns).Get(ctx, jobName, metav1.GetOptions{})
-		if apierrors.IsNotFound(err) {
-			return true, nil
-		}
-		return false, err
-	})
+	return framework.Gomega().
+		Eventually(ctx, func(ctx context.Context) error {
+			_, err := c.BatchV1().Jobs(ns).Get(ctx, jobName, metav1.GetOptions{})
+			return err
+		}).
+		WithPolling(framework.Poll).
+		WithTimeout(timeout).
+		Should(gomega.MatchError(apierrors.IsNotFound, fmt.Sprintf("that job %s/%s should be gone", ns, jobName)))
```

Comment:
```suggestion
		Should(gomega.MatchError(apierrors.IsNotFound, fmt.Sprintf("expected job %q to be gone", klog.KRef(ns, jobName))))
```

Reply:
```
func (matcher *MatchErrorMatcher) FailureMessage(actual interface{}) (message string) {
	if matcher.isFunc {
		return format.Message(actual, fmt.Sprintf("to match error function %s", matcher.FuncErrDescription[0]))
	}
	return format.Message(actual, "to match error", matcher.Expected)
}
```

https://github.com/kubernetes/kubernetes/blob/framework-job/vendor/github.com/onsi/gomega/matchers/match_error_matcher.go#L74-L80

To make the whole error message more readable, we can add the `that` word to the FuncErrDescription string. @kaisoz